src/xcsync/SyncContext.cs

```diff
@@ -500,7 +500,7 @@ async Task SyncFromXcodeAsync (CancellationToken token)
 			await (syncItem switch {
 #pragma warning disable CA2012 // Use ValueTasks correctly  
 				SyncableType type => Hub.PublishAsync (SyncChannel, new LoadTypesFromObjCMessage (Guid.NewGuid ().ToString (), tcs = new TaskCompletionSource (), xcodeWorkspace, syncItem)),
-				SyncableContent file => Hub.PublishAsync (FileChannel, new CopyFileMessage (Guid.NewGuid ().ToString (), file.SourcePath, FileSystem.Path.Combine (ProjectDir!, FileSystem.Path.Combine (basePath, file.DestinationPath)))),
+				SyncableContent file => Hub.PublishAsync (FileChannel, new CopyFileMessage (Guid.NewGuid ().ToString (), file.SourcePath, FileSystem.Path.Combine (ProjectDir!, FileSystem.Path.Combine (basePath, FileSystem.Path.GetRelativePath (basePath, file.DestinationPath))))),
```

Comment:
1. Is there a test for this?
2. Was `file.DestinationPath` ever correct usage here?
3. Could `file.DestinationPath` ever be a full path?

Reply:
there are tests for sync functionality, but this specifically stemmed from an issue of whether a full vs. relative project path is used. these are the types of scenarios I hope to have CSI vendors to help us out with